ci/pipeline/ray_coverage.py

```diff
@@ -21,9 +32,24 @@ def main(test_target: str) -> None:
     _run_test(test_target, coverage_file)
     coverage_info = _collect_coverage(coverage_file)
     logger.info(coverage_info)
+    if upload:
+        s3_file_name = _persist_coverage_info(coverage_file)
+        logger.info(f"Successfully uploaded coverage data to s3 as {s3_file_name}")
```

Comment:
just move this inside the function? and then you do not need to return the `s3_file_name`. returning the `s3_file_name` feels like leaking internal info.